app/src/androidTest/java/com/github/se/polyfit/data/local/database/MealDatabaseTest.kt

```diff
@@ -128,6 +130,51 @@ class MealDatabaseTest {
     assertContains(ingredients, ingredientList.first())
   }
 
+  @Test
+  fun testGetAllMealsFromACertainDate() {
+    mealDao.deleteAll()
+
+    val meal = createMeal("Oatmeal", "1", LocalDate.now())
+    val meal2 = createMeal("Chocolate", "2", LocalDate.now())
+    val meal3 = createMeal("Lobster", "3", LocalDate.now().minusWeeks(1))
+
+    mealDao.insert(meal)
+    mealDao.insert(meal2)
+    mealDao.insert(meal3)
+
+    val meals = mealDao.getMealsCreatedOnOrAfterDate(LocalDate.now())
+
+    assert(meals.size == 2)
+    assert(meals.contains(meal))
+    assert(meals.contains(meal2))
+    assertFalse(meals.contains(meal3))
+
+    val newMeals = mealDao.getMealsCreatedOnOrAfterDate(LocalDate.now().minusMonths(1))
+
+    assert(newMeals.contains(meal3))
+  }
+
+  private fun createMeal(name: String, firebaseId: String, createdAt: LocalDate): Meal {
+    return Meal(
+        name = name,
+        occasion = MealOccasion.BREAKFAST,
+        mealID = 1,
+        mealTemp = 20.0,
+        nutritionalInformation =
```

Comment:
i don't think it affects this test, but in general after that bugfix I've learned it is better to create nutrition through your ingredients rather than directly in the meal. perhaps we should even make it not possible to do so

Reply:
I see what you mean. The auto-update feature of meals could cause the recalculation of `nutritionalInformation` to be off from what is expected. We could fix this in a later PR by setting the `nutritionalInformation` to empty by default, to be sure that the correct values are automatically calculated.